lib/services/extensibility-service.ts

```diff
@@ -0,0 +1,135 @@
+
+		await this.assertExtensionsSetup();
+
+		await this.$npm.uninstall(extensionName, { save: true }, this.pathToExtensions);
+
+		this.$logger.trace(`Finished uninstallation of extension '${extensionName}'.`);
+	}
+
+	public loadExtensions(): Promise<any>[] {
+		this.$logger.trace("Loading extensions.");
+
+		let dependencies: IStringDictionary = null;
+
+		try {
+			dependencies = this.getInstalledExtensions();
+		} catch (err) {
+			this.$logger.trace(`Error while getting installed dependencies: ${err.message}. No extensions will be loaded.`);
+		}
+
+		return _.keys(dependencies)
+			.map(name => this.loadExtension(name));
+	}
+
+	public getInstalledExtensions(): IStringDictionary {
+		if (this.$fs.exists(this.pathToPackageJson)) {
+			return this.$fs.readJson(this.pathToPackageJson).dependencies;
+		}
+
+		return null;
+	}
+
+	private async loadExtension(extensionName: string): Promise<IExtensionData> {
+		try {
+			await this.assertExtensionIsInstalled(extensionName);
+
+			const pathToExtension = path.join(this.pathToExtensions, constants.NODE_MODULES_FOLDER_NAME, extensionName);
+			this.$requireService.require(pathToExtension);
+			return { extensionName };
+		} catch (error) {
+			this.$logger.warn(`Error while loading ${extensionName} is: ${error.message}`);
+			const err = <IExtensionLoadingError>new Error(`Unable to load extension ${extensionName}. You will not be able to use the functionality that it adds.`);
+			err.extensionName = extensionName;
+			throw err;
+		}
+	}
+
+	private async assertExtensionIsInstalled(extensionName: string): Promise<void> {
+		this.$logger.trace(`Asserting extension ${extensionName} is installed.`);
+		const installedExtensions = this.$fs.readDirectory(path.join(this.pathToExtensions, constants.NODE_MODULES_FOLDER_NAME));
+
+		if (installedExtensions.indexOf(extensionName) === -1) {
+			this.$logger.trace(`Extension ${extensionName} is not installed, starting installation.`);
+			await this.installExtension(extensionName);
+		}
+
+		this.$logger.trace(`Extension ${extensionName} is installed.`);
+	}
+
+	@cache()
+	private async assertExtensionsSetup(): Promise<void> {
+		this.assertExtensionsDirExists();
+		this.assertPackageJsonExists();
+	}
+
+	@cache()
+	private assertExtensionsDirExists(): void {
+		if (!this.$fs.exists(this.pathToExtensions)) {
+			this.$fs.createDirectory(this.pathToExtensions);
+		}
+	}
+
+	@cache()
+	private assertPackageJsonExists(): void {
+		if (!this.$fs.exists(this.pathToPackageJson)) {
+			this.$logger.trace(`Creating ${this.pathToPackageJson}.`);
+
+			// create default package.json
+			this.$fs.writeJson(this.pathToPackageJson, {
+				name: "nativescript-extensibility",
+				version: "1.0.0",
+				description: "The place where all packages that extend CLI will be installed.",
+				license: "Apache-2.0",
+				readme: "The place where all packages that extend CLI will be installed.",
+				repository: "none",
+				dependencies: {}
```

Comment:
You can skip writing a lot of the keys for example: the `license`, `repository` and `dependencies` are gratuitous IMO